cmd/operator/main.go

```diff
@@ -95,40 +114,48 @@ func main() {
 
 	withWebhook := true
 
-	// TODO(mythi): add a StringVar flag to select which controllers to enable
-	for _, device := range []string{"dsa", "fpga", "gpu", "qat", "sgx"} {
+	for device, enabled := range devices {
+		if !enabled {
+			continue
+		}
 		if err = setupControllerAndWebhook[device](mgr, ns, withWebhook); err != nil {
 			setupLog.Error(err, "unable to initialize controller", "controller", device)
 			os.Exit(1)
 		}
 	}
 
-	pm := patcher.NewPatcherManager(mgr.GetLogger().WithName("webhooks").WithName("Fpga"))
-
-	mgr.GetWebhookServer().Register("/pods", &webhook.Admission{
-		Handler: admission.HandlerFunc(pm.GetPodMutator()),
-	})
+	if devices["fpga"] {
+		pm = patcher.NewPatcherManager(mgr.GetLogger().WithName("webhooks").WithName("Fpga"))
 
-	mgr.GetWebhookServer().Register("/pods-sgx", &webhook.Admission{
-		Handler: &sgxwebhook.SgxMutator{Client: mgr.GetClient()},
-	})
+		mgr.GetWebhookServer().Register("/pods", &webhook.Admission{
+			Handler: admission.HandlerFunc(pm.GetPodMutator()),
+		})
+	}
 
-	if err = (&fpgacontroller.AcceleratorFunctionReconciler{
-		Client:         mgr.GetClient(),
-		Scheme:         mgr.GetScheme(),
-		PatcherManager: pm,
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "AcceleratorFunction")
-		os.Exit(1)
+	if devices["sgx"] {
+		mgr.GetWebhookServer().Register("/pods-sgx", &webhook.Admission{
+			Handler: &sgxwebhook.SgxMutator{Client: mgr.GetClient()},
+		})
 	}
 
-	if err = (&fpgacontroller.FpgaRegionReconciler{
-		Client:         mgr.GetClient(),
-		Scheme:         mgr.GetScheme(),
-		PatcherManager: pm,
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "FpgaRegion")
-		os.Exit(1)
+	if devices["fpga"] {
```

Comment:
this if condition and above ```if devices["fpga"]``` can be merged into one. The same can be done for 2 ```if devices["sgx"]``` conditions.

Reply:
OK, thanks! I will merge them.